lib/protobuf/field/base_field.rb

```diff
@@ -176,16 +176,18 @@ def define_accessor
         else
           define_getter
           define_setter
+          define_decode_setter
         end
       end
 
       def define_array_getter
         field = self
+        field_name = field.name
         method_name = field.getter
 
         message_class.class_eval do
           define_method(method_name) do
-            @values[field.name] ||= ::Protobuf::Field::FieldArray.new(field)
+            @values[field_name] ||= ::Protobuf::Field::FieldArray.new(field)
```

Comment:
Heads up that a lot of this work will need to be moved to the `[]=` and `[]` methods after PR #302 is merged. In fact a lot of the setter/getter code in this PR will conflict, I comment on just a little of it below.